core/payload/deploy_code.go

```diff
@@ -54,6 +65,30 @@ func (dc *DeployCode) Address() common.Address {
 	return dc.address
 }
 
+func (dc *DeployCode) SetVmType(ty VmType) {
+	dc.vmFlags = byte(ty)
+}
```

Comment:
check ty here?

Reply:
ty is `VmType`, which is guranteed to be valid.